credentials/alts/internal/conn/record.go

```diff
@@ -153,20 +170,30 @@ func (p *conn) Read(b []byte) (n int, err error) {
 		// Check whether the next frame to be decrypted has been
 		// completely received yet.
 		if len(framedMsg) == 0 {
-			copy(p.protected, p.nextFrame)
-			p.protected = p.protected[:len(p.nextFrame)]
+			p.protected = p.protected[:copy(p.protected, p.nextFrame)]
```

Comment:
I think this line is a little challenging to reason about since it's modifying `p.protectd` in several ways. Can it be broken up without performance hits, or will not immediately slicing to the right length cause an issue?